src/main/java/sparksoniq/jsoniq/runtime/iterator/functions/SubstringFunctionIterator.java

```diff
@@ -0,0 +1,52 @@
+package sparksoniq.jsoniq.runtime.iterator.functions;
+
+import sparksoniq.exceptions.IteratorFlowException;
+import sparksoniq.exceptions.SparksoniqRuntimeException;
+import sparksoniq.jsoniq.item.IntegerItem;
+import sparksoniq.jsoniq.item.Item;
+import sparksoniq.jsoniq.item.StringItem;
+import sparksoniq.jsoniq.item.metadata.ItemMetadata;
+import sparksoniq.jsoniq.runtime.iterator.RuntimeIterator;
+import sparksoniq.jsoniq.runtime.iterator.functions.base.LocalFunctionCallIterator;
+import sparksoniq.jsoniq.runtime.metadata.IteratorMetadata;
+
+import java.util.List;
+
+public class SubstringFunctionIterator extends LocalFunctionCallIterator {
+    public SubstringFunctionIterator(List<RuntimeIterator> arguments, IteratorMetadata iteratorMetadata) {
+        super(arguments, iteratorMetadata);
+        if (arguments.size() < 2 || arguments.size() > 3)
```

Comment:
This is also the kind of checks that will no longer be needed at runtime with a function catalogue.